docs/azure/sdk/azure-sdk-thread-safety.md

```diff
@@ -0,0 +1,124 @@
+
+```csharp
+var client = new SecretClient(new Uri("<secrets_endpoint>"), new DefaultAzureCredential());
+
+foreach (var secretName in secretNames)
+{
+    // Using clients from parallel threads
+    Task.Run(() => Console.WriteLine(client.GetSecret(secretName).Value));
+}
+```
+
+Model objects used by SDK clients, whether input or output models are *not* thread-safe by default.  Most use cases involving model objects only involve a single thread, making the cost of implementing synchronization as a default behavior too high for these objects. The following sample illustrates a bug where accessing a model from multiple threads could cause an undefined behavior.
+
+```csharp
+KeyVaultSecret newSecret = client.SetSecret("secret", "value");
+
+foreach (var tag in tags)
+{
+    // Don't use model type from parallel threads
+    Task.Run(() => newSecret.Properties.Tags[tag] = CalculateTagValue(tag));
+}
+
+client.UpdateSecretProperties(newSecret.Properties);
+
+```
+
+If you need to access the model from different threads, you must implement your own synchronization code.
+
+```csharp
+KeyVaultSecret newSecret = client.SetSecret("secret", "value");
+
+foreach (var tag in tags)
+{
+    Task.Run(() =>
+    {
+        lock (newSecret)
+        {
+            newSecret.Properties.Tags[tag] = CalculateTagValue(tag);
+        }
+    );
+}
+
+client.UpdateSecretProperties(newSecret.Properties);
+```
+
+## Client Lifetime
+
+Because Azure SDK clients are thread-safe, there is no reason to construct multiple SDK client objects for a given set of constructor parameters. Therefore, you should treat Azure SDK client objects as singletons once constructed.  This is most commonly implemented by registering Azure SDK client objects as singletons in the application's IoC container and using dependency injection to pass out references to the SDK client object.  The following example shows how this is done in an ASP.NET Core application with the built-in IoC container.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddControllersWithViews();
+
+    var blobServiceClient = new BlobServiceClient(new Uri("<secrets_endpoint>"), new DefaultAzureCredential());                   
+    services.AddSingleton(blobServiceClient);
+          
+    // Additional code ...
+}
+
+```
+
+For more information about implementing dependency injection with the Azure SDK, refer to the article [Dependency injection with the Azure .NET SDK](./dependency-injection.md).
+
+Alternatively, you may also create the instance of an SDK client and pass it around as a parameter to methods in need of a client. The point is, avoid unnecessary instantiations of the same SDK client object with the same parameters as it's unnecessary and wasteful.
```

Comment:
```suggestion
Alternatively, you may also create an SDK client instance and pass it as a parameter to methods that require a client. The point is to avoid unnecessary instantiations of the same SDK client object with the same parameters, as it's unnecessary and wasteful.
```